src/Graph.ts

```diff
@@ -437,7 +434,9 @@ export default class Graph {
 
 		const visit = (module: Module, seen: { [id: string]: boolean } = {}) => {
 			if (seen[module.id]) {
-				hasCycles = true;
+				if (!allSeen[module.id]) {
+					this.warnCycle(module);
```

Comment:
For even better debugging perhaps pass the full cycle stack?

```js
let cycleStack = [];
Object.keys(seen).forEach(name => {
  if (cycleStack) {
    cycleStack.push(name);
  } else if (name === module.id) {
    cycleStack = [];
  }
});
this.warnCycle(module, cycleStack);
```

Then something like `stack: cycleStack.join(' -> ')` in the log.

Reply:
Ideally the above should be relative to the base like we do in other places to be readable.

Reply:
Sorry, being silly here, this isn't a parent-based structure so doesn't capture the path of course.

Reply:
I'll try to get the full cycle. I have an idea how to do that :)

Reply:
`relativeId` strips `\0000commonjs-proxy:` prefixes :(